pkg/logs/tag/provider.go

```diff
@@ -62,7 +62,8 @@ func (p *provider) GetTags() []string {
 
 		// start timer if necessary
 		go func() {
```

Comment:
nit: I think we should start this goroutine only if `p.submitExpectedTags` is true